### ISSUE TITLE
How to subscribe to model instance creation?

### ISSUE THREAD

**Issue by zhitnik-hiteklab** [hishnash/djangochannelsrestframework#63]
Sorry for writing here, but I can't find answer in docs. Client part of my app expects backend to create new model instance. If I use `subscribe_instance`, I get 404 response right after subscription, because the instance is not exist yet. The client have to watch for an instance with exact pk... I'm new in channels, maybe this task is relatively simple.

**Comment by HtheChemist** [comment]
When you make a POST call on the API to create a new model instance, it will return a 201 response with the information including the pk or id of the created instance. Depending on what will receive this response you can easily parse it to get the id of the created instance and pass it to you subscriber. It is not important if the pk is sequential or not, since it will be included in the response.

If you are not the one doing the API call to create the instance, in this case you can subscribe to the entire model with a consumer or serializer that will pass the new instance id if the user has the right permissions, and subscribe to it.

So in summary:

If you are the one doing the API call creating the new instance:
API POST -> 201 Response -> Parse the id from the response -> Subscribe to the instance

If you are not the one doing the API call creating the new instance:
Subscribe to the model -> Pass the message through a Serializer to ensure permission in a consumer view -> Push the pk to the front end -> Subscribe to the instance

**Comment by zhitnik-hiteklab** [comment]
Ok, I made it via custom actions, like in the last example:
```python
class PhoneConsumer(AsyncAPIConsumer):
    serializer_class = PhoneSerializer

    @model_observer(Phone)
    async def phone_change_handler(self, message, observer=None, action=None, **kwargs):
        if action != 'create':
            return
        message['action'] = action
        await self.send_json(message)

    @phone_change_handler.groups_for_signal
    def phone_change_handler(self, instance: Phone, **kwargs):
        yield f'-token__{instance.token_id}'

    @phone_change_handler.groups_for_consumer
    def phone_change_handler(self, token, **kwargs):
        yield f'-token__{token}'

    @action()
    async def subscribe_to_phone_create(self, token=None, **kwargs):
        if token is None:
            raise ValueError("token must have a value set")
        await self.phone_change_handler.subscribe(token=token)

    @action()
    async def unsubscribe_to_phone_create(self, token=None, **kwargs):
        if token is None:
            raise ValueError("token must have a value set")
        await self.phone_change_handler.unsubscribe(token=token)
```
Thanks for your help!
